src/components/DonatePage/DonatePageBody.tsx

```diff
@@ -155,6 +166,11 @@ const DonatePageBody = () => {
       });
 
       setHasSavedDonation(true);
+
+      // Save that user accepted terms and conditions
+      if (terms && !user?.acceptedToS) {
```

Comment:
I don't know when is it in the flow, but I think the best place is calling it before making the tx or saving the draft donation!